tests/validate_json_format.py

```diff
@@ -169,20 +174,40 @@ def log_failure(self, code, line=None):
         self.fail = True
 
     def check_format(self, file_name):
-        """Initiate teh check."""
+        """Initiate the check."""
 
         self.fail = False
+        comment_align = None
         with codecs.open(file_name, encoding='utf-8') as f:
             count = 1
             for line in f:
+                indent_match = (RE_LINE_INDENT_TAB if self.use_tabs else RE_LINE_INDENT_SPACE).match(line)
```

Comment:
You like nested if else blocks? 

How do you think about `indent_match = RE_LINE_INDENT[self.use_tabs].match(line)`
and defining  `RE_LINE_INDENT = [RE_LINE_INDENT_SPACE, RE_LINE_INDENT_TAB]` after line 49

Reply:
I hate 80 char line enforcement.  I think they are outdated.  I stick to 120.  80 came about from the era of punch cards, because that is how many holes you could punch on a line.  That carried over into terminals and editors, but in the age of widescreen HDPI, I think people are holding onto that number because they feel they need to without knowing why.  I have no idea why every linter in the world defaults to 80 😕.

No offence if you are an 80 char line guy :slightly_smiling_face:.

As for this particular line, I'm indifferent.  If it goes past 120 I'll refactor, but if not, meh.

Reply:
I have no problem with longer lines, some projects enforce it. I had a hard discussion with other contributors on GitGutter about it - terrible.

The "advantages" I see with 80 chars is:
- it makes you think more thoroughly what to write
- causes you to split things more often which in the end makes code more clearly to read
- some very long line with many nested functions are really hard to read sometimes
- limited line length is good if it comes to have 2 or 3 files next to each other on a 3-way-diff, which will become obsolete with latest 37`` super-wide-screens 😄 

But I agree with 80 to be too small to handle 3rd or 4th indention level and expressive variable/function names.

For some reason all the linters use 80 by default.

Reply:
I've posted URLs in comments longer than 80 chars :slightly_smiling_face: .

By no means am I discounting the use of 80 chars.  There are a lot of decent, valid arguments for it.  There are also plenty valid ones against it as well.

I guess in general, when I code, I lean towards what annoys me less.  I think 120 is the perfect amount to get out a complete thought without annoying me with constant breaks.  Even if screens get 4x greater, I think 120 is my thinking limit.  It has to be small enough I can take the full line in, but large enough to get out a reasonable thought with expressive variable names.

It's like when you get into projects that break things up into so many tiny functions that it can become difficult to get an overview of what the file is doing.  And conversely, functions that are too big cause you to get lost.

Reply:
Yep, it's a balancing act.